src/failed-transactions/failed-transactions.js

```diff
@@ -0,0 +1,104 @@
+const {
+  Finding, FindingSeverity, FindingType,
+} = require('forta-agent');
+
+// addresses we are interested in monitoring
+const accountAddresses = require('../../account-addresses.json');
+const config = require('../../agent-config.json');
+
+const initializeData = {};
+
+// formats provided data into a Forta alert
+function createAlert(name, address, failedTxs, blockWindow, everestId) {
+  return Finding.fromObject({
+    name: 'Perp.Fi Failed Transactions Alert',
+    description: `${name} has sent ${failedTxs.length} failed transactions `
+    + `in the past ${blockWindow} blocks`,
+    protocol: 'Perp.Fi',
+    alertId: 'AE-PERPFI-FAILED-TRANSACTIONS',
+    severity: FindingSeverity.Medium,
+    type: FindingType.Info,
+    everestId,
+    metadata: {
+      name,
+      address,
+      failedTxs,
+    },
+  });
+}
+
+function provideHandleTransaction(data) {
+  /**
+   * data is expected to have:
+   *  - addresses (object of name:address entries we are interested in)
+   *  - failedTxs (object to hold failed txs)
+   *  - blockWindow (config of how many blocks we should look for failed txs)
+   *  - failedTxLimit (how many failed txs required to raise an alert)
+   *  - everestId (project ID from https://everest.link)
+   */
+  return async function handleTransaction(txEvent) {
+    if (!data) throw new Error("called handler before initializing");
```

Comment:
I ran this in a Node session and did not see any output:
```javascript
data = {};
if (!data) {
  console.log('Will we ever see this?');
}
```
Are we hoping to check for `data === undefined`?  Or are we also hoping to check for `data === {}`?

Reply:
fantastic point